Finances

msoucy opened the pull request:
Closes #1

bot190 commented on signinapp/model.py:
```diff
@@ -26,6 +27,9 @@
     normalize_phone_number_from_storage,
 )
 
+FINANCE_OVERHEAD = 0.15
```
This should be stored per event so we can account for actual costs of running an event.

bot190 reviewed (commented):
Looking good overall, I think we want to be a bit more flexible with the overhead costs though.

msoucy commented:
Still needs a few things, but it's a start

- [x] Way to get total earned by user
- [x] Way to mark a user as being eligible to receive funds (don't want mentors counting towards fundraising hours)
- [x] Way to show total remaining funds for a user (OBE - we're not using this as the final source of financial information)
- [x] Better overhead management

msoucy commented:
Overhead notes:

- [x] Track how much it costs to run each event? Runs into risk of making things too complicated. But things like FLL, Mayhem, pasta dinners, all have up front costs.
- [x] Original idea of overhead was to track funds that go to the team as a whole instead of the students. Number might need to have a default (maybe even default per event type) and be per event - percentage vs total? Estimated default at 50%
- [x] formula using overhead makes sense
- [x] mentors don't count for fund earning
- [x] way to track total overhead earned
- [x] finance page showing overhead and student funds

bot190 commented on signinapp/events.py:
```diff
@@ -51,6 +52,12 @@ class EventForm(FlaskForm):
+    cost = DecimalField(label="Event Cost", filters=[lambda x: int(x * 100)])
+    overhead = DecimalField(
+        label="Overhead Percentage",
+        validators=[NumberRange(min=0.0, max=1.0, message="Must be a percentage")],
```
If this is a percentage shouldn't it be 0-100?

bot190 commented on signinapp/model.py:
```diff
@@ -442,6 +481,26 @@ def is_active(cls):
+
+    @property
+    def overhead_funds(self) -> str:
+        return locale.currency(self.funds * self.overhead / 100.0)
```
```suggestion
        return locale.currency(self.net_funds * self.overhead / 100.0)
```
This should account for the event cost

bot190 commented on signinapp/events.py:
```diff
@@ -51,6 +52,12 @@ class EventForm(FlaskForm):
     end = DateTimeLocalField(format=DATE_FORMATS)
     type_id = SelectField(label="Type", choices=lambda: get_form_ids(EventType))
     enabled = BooleanField(default=True)
+    funds = DecimalField(label="Funds Received", filters=[lambda x: int(x * 100)])
```
These filters seem to cause the funds to grow every time you edit the page (they also currently break creating an event because it tries to apply the filter to a `None` value.

bot190 reviewed (changes requested):
Overall this looks good. We should probably talk with Matt about this and see what he thinks.

msoucy committed:
Add financial support

Closes #1

bot190 commented on signinapp/events.py:
```diff
@@ -244,6 +256,8 @@ def edit_event():
         form.start.data = correct_time_for_storage(form.start.data)
         form.end.data = correct_time_for_storage(form.end.data)
         form.populate_obj(event)
+        event.cost = int(form.cost.data / 100)
```
Each time you edit an event all of the money values are divided by 100. Seems like there's a missing multiplication somewhere here.

bot190 committed:
fix: Convert dollar inputs to cents for storage properly 

Initialize finance blueprint following new style
Multiply dollar values on input to store as cents
Fix overhead calculation on finance page to account for event cost

bot190 reviewed (approved):
I pushed some fixes to get the money values presented correctly, as well as initialize the finance blueprint. Lets get this merged!
